docs/tutorial.md

```diff
@@ -0,0 +1,507 @@
+          return when (screen) {
+            is DetailScreen -> return DetailPresenter(screen, emailRepository)
+            else -> null
+          }
+        }
+      }
+    }
+    ```
+
+Here we have access to the screen and dynamically create the presenter we need. It can then pass the screen on to the presenter.
+
+We can then wire these detail components to our `Circuit` instance.
+
+=== "Circuit instance"
+    ```kotlin
+    val circuit: Circuit =
+      Circuit.Builder()
+        // ...
+        .addPresenterFactory(DetailPresenter.Factory(emailRepository))
+        .addUi<DetailScreen, DetailScreen.State> { state, modifier -> EmailDetail(state, modifier) }
+        .build()
+    ```
+
+## Navigate to the detail screen
+
+Now that we have a detail screen, let's navigate to it from our inbox list. As you can see in our presenter factory above, Circuit also offers access to a `Navigator` in this `create()` call that factories can then pass on to their created presenters. 
+
+Let's add a `Navigator` property to our presenter and create a factory for our inbox screen now.
+
+=== "InboxPresenter"
+    ```kotlin
+    class InboxPresenter(
+      private val navigator: Navigator,
+      private val emailRepository: EmailRepository
+    ) : Presenter<InboxScreen.State> {
+      // ...
+      class Factory(private val emailRepository: EmailRepository) : Presenter.Factory {
+        override fun create(screen: Screen, navigator: Navigator, context: CircuitContext): Presenter<*>? {
+          return when (screen) {
+            InboxScreen -> return InboxPresenter(navigator, emailRepository)
+            else -> null
+          }
+        }
+      }
+    }
+    ```
+
+=== "Circuit instance"
+    ```kotlin
+    val circuit: Circuit =
+      Circuit.Builder()
+        .addPresenterFactory(InboxPresenter.Factory(emailRepository))
+        .addUi<InboxScreen, InboxScreen.State> { state, modifier -> Inbox(state, modifier) }
+        .addPresenterFactory(DetailPresenter.Factory(emailRepository))
+        .addUi<DetailScreen, DetailScreen.State> { state, modifier -> EmailDetail(state, modifier) }
+        .build()
+    ```
+
+Now that we have a `Navigator` in our inbox presenter, we can use it to navigate to the detail screen. First, we need to explore how events work in Circuit.
+
+## Events
+
+So far, we've only covered _state_. State is produced by the presenter and consumed by the UI. That's only half of the UDF picture though! Events are the opposite: they're produced by the UI and consumed by the presenter. Events are how you can trigger actions in your app, such as navigation. _This completes the circuit._
+
+Events in Circuit are a little unconventional in Circuit doesn't actually provide first-party infrastructure for pipelining events from the UI to presenters. Instead, we use an _event sink property_ pattern, where states contain a trailing `eventSink` function that accepts events and can be emitted to by the UI.
```

Comment:
First sentence here is unclear

Reply:
[c44b252](https://github.com/slackhq/circuit/pull/1158/commits/c44b25248412c7e266867188b84c4b9ee1cebba9)